operator/auth/spire/client.go

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package spire
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	entryv1 "github.com/spiffe/spire-api-sdk/proto/spire/api/server/entry/v1"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/cilium/cilium/operator/auth/identity"
+	"github.com/cilium/cilium/pkg/backoff"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+const (
+	notFoundError   = "NotFound"
+	defaultParentID = "/dclient"
+	pathPrefix      = "/cilium-id"
+)
+
+var initializedError = fmt.Errorf("unable to connect to SPIRE server")
```

Comment:
Seems like it might be nice to have this include the configured details? (ie `spire-server.spire.svc.cluster.local` on port whatever)

Reply:
ah good point :100: